### ISSUE TITLE
test::executable_test' panicked

### ISSUE THREAD

**Issue by rossmuir** [maidsafe/maidsafe_vault#100]
CI failing `test::executable_test' ` on all 3 platforms

```
maidsafe@SAFE-RUST-SEED:~/maidsafe_vault$ RUST_TEST_THREADS=1 cargo test
   Compiling maidsafe_vault v0.0.3 (file:///home/maidsafe/maidsafe_vault)
     Running target/debug/maidsafe_vault-1f95d1bcce89e4ec

running 38 tests
test chunk_store::chunk_store::dummy ... ok
test chunk_store::dummy_cs ... ok
test chunk_store::test::test::constructor_initialization ... ok
test chunk_store::test::test::put_and_get_value_should_be_same ... ok
test chunk_store::test::test::remove_from_disk_store ... ok
test chunk_store::test::test::repeatedly_storing_same_name ... ok
test chunk_store::test::test::should_fail_if_chunk_size_is_greater_than_max_disk_size ... ok
test chunk_store::test::test::should_fail_on_disk_overfill ... ok
test chunk_store::test::test::successful_store ... ok
test data_manager::database::test::exist ... ok
test data_manager::database::test::handle_account_transfer ... ok
test data_manager::database::test::put ... ok
test data_manager::database::test::remove_pmid ... ok
test data_manager::database::test::replace_pmids ... ok
test data_manager::test::handle_account_transfer ... ok
test data_manager::test::handle_put_get ... ok
test data_manager::test::handle_stats_transfer ... ok
test maid_manager::database::test::delete_data ... ok
test maid_manager::database::test::exist ... ok
test maid_manager::database::test::handle_account_transfer ... ok
test maid_manager::database::test::maid_manager_account_wrapper_serialisation ... ok
test maid_manager::database::test::put_data ... ok
test maid_manager::test::handle_account_transfer ... ok
test maid_manager::test::handle_put ... ok
test pmid_manager::database::test::exist ... ok
test pmid_manager::database::test::handle_account_transfer ... ok
test pmid_manager::database::test::pmid_manager_account_serialisation ... ok
test pmid_manager::test::handle_account_transfer ... ok
test pmid_manager::test::handle_put ... ok
test pmid_node::test::handle_put_get ... ok
test test::executable_test ... NOTE: Transport connect 45.55.207.180:34696 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:34696 failure due to Connection refused (os error 111)
thread '<main>' panicked at 'assertion failed: cm.send(bootstrapped_to.clone(), serialised_message).is_ok()', /home/maidsafe/.cargo/registry/src/github.com-0a35038f75765ae4/routing-0.1.64/src/routing_node.rs:159
NOTE: Transport connect 127.0.0.1:34696 failure due to Connection refused (os error 111)
NOTE: Transport connect 45.55.207.180:34696 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:50820 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:34696 failure due to Connection refused (os error 111)
NOTE: Transport connect 45.55.207.180:50820 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:50820 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:34696 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:34696 failure due to Connection refused (os error 111)
NOTE: Transport connect 45.55.207.180:34696 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:34696 failure due to Connection refused (os error 111)
NOTE: Transport connect 45.55.207.180:50820 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:50820 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:50820 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:34696 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:36048 failure due to Connection refused (os error 111)
thread '<main>' panicked at 'assertion failed: cm.send(bootstrapped_to.clone(), serialised_message).is_ok()', /home/maidsafe/.cargo/registry/src/github.com-0a35038f75765ae4/routing-0.1.64/src/routing_node.rs:159
NOTE: Transport connect 127.0.0.1:41301 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:41301 failure due to Connection refused (os error 111)
NOTE: Transport connect 45.55.207.180:41301 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:36048 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:34696 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:34696 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:34696 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:50820 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:50820 failure due to Connection refused (os error 111)
NOTE: Transport connect 45.55.207.180:50820 failure due to Connection refused (os error 111)
NOTE: Transport connect 45.55.207.180:34696 failure due to Connection refused (os error 111)
thread '<main>' panicked at 'assertion failed: cm.send(bootstrapped_to.clone(), serialised_message).is_ok()', /home/maidsafe/.cargo/registry/src/github.com-0a35038f75765ae4/routing-0.1.64/src/routing_node.rs:159
NOTE: Transport connect 127.0.0.1:34696 failure due to Connection refused (os error 111)
NOTE: Transport connect 45.55.207.180:34696 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:34696 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:50820 failure due to Connection refused (os error 111)
NOTE: Transport connect 45.55.207.180:50820 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:50820 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:34696 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:36048 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:41301 failure due to Connection refused (os error 111)
NOTE: Transport connect 45.55.207.180:41301 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:41301 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:37288 failure due to Connection refused (os error 111)
NOTE: Transport connect 45.55.207.180:37288 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:37288 failure due to Connection refused (os error 111)
thread '<main>' panicked at 'assertion failed: cm.send(bootstrapped_to.clone(), serialised_message).is_ok()', /home/maidsafe/.cargo/registry/src/github.com-0a35038f75765ae4/routing-0.1.64/src/routing_node.rs:159
NOTE: Transport connect 127.0.0.1:34696 failure due to Connection refused (os error 111)
NOTE: Transport connect 45.55.207.180:34696 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:34696 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:50820 failure due to Connection refused (os error 111)
NOTE: Transport connect 45.55.207.180:50820 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:50820 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:34696 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:36048 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:41301 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:41301 failure due to Connection refused (os error 111)
NOTE: Transport connect 45.55.207.180:41301 failure due to Connection refused (os error 111)
NOTE: Transport connect 45.55.207.180:37288 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:37288 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:37288 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:59164 failure due to Connection refused (os error 111)
NOTE: Transport connect 45.55.207.180:59164 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:59164 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:34696 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:50820 failure due to Connection refused (os error 111)
NOTE: Transport connect 45.55.207.180:34696 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:34696 failure due to Connection refused (os error 111)
NOTE: Transport connect 45.55.207.180:50820 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:50820 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:34696 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:36048 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:41301 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:41301 failure due to Connection refused (os error 111)
NOTE: Transport connect 45.55.207.180:41301 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:37288 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:37288 failure due to Connection refused (os error 111)
NOTE: Transport connect 45.55.207.180:37288 failure due to Connection refused (os error 111)
NOTE: Transport connect 45.55.207.180:59164 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:59164 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:59164 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:41469 failure due to Connection refused (os error 111)
NOTE: Transport connect 45.55.207.180:41469 failure due to Connection refused (os error 111)
FAILED
test test::lib_test ... NOTE: Transport connect [::1]:34228 failure due to Connection refused (os error 111)
BOOTSTRAP to Tcp(V4(127.0.0.1:34228))
NODE listening on Some(Tcp(V4(127.0.0.1:36230)))
thread '<unnamed>' panicked at 'assertion failed: cm.send(bootstrapped_to.clone(), serialised_message).is_ok()', /home/maidsafe/.cargo/registry/src/github.com-0a35038f75765ae4/routing-0.1.64/src/routing_node.rs:159
NOTE: Transport connect [::1]:34228 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:36230 failure due to Connection refused (os error 111)
BOOTSTRAP to Tcp(V4(45.55.207.180:34228))
NODE listening on Some(Tcp(V4(127.0.0.1:50704)))
thread '<unnamed>' panicked at 'assertion failed: cm.send(bootstrapped_to.clone(), serialised_message).is_ok()', /home/maidsafe/.cargo/registry/src/github.com-0a35038f75765ae4/routing-0.1.64/src/routing_node.rs:159
NOTE: Transport connect [::1]:34228 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:36230 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:50704 failure due to Connection refused (os error 111)
BOOTSTRAP to Tcp(V4(127.0.0.1:34228))
NODE listening on Some(Tcp(V4(127.0.0.1:34818)))
thread '<unnamed>' panicked at 'assertion failed: cm.send(bootstrapped_to.clone(), serialised_message).is_ok()', /home/maidsafe/.cargo/registry/src/github.com-0a35038f75765ae4/routing-0.1.64/src/routing_node.rs:159
NOTE: Transport connect [::1]:34228 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:36230 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:50704 failure due to Connection refused (os error 111)
NOTE: Transport connect 45.55.207.180:50704 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:50704 failure due to Connection refused (os error 111)
BOOTSTRAP to Tcp(V4(127.0.0.1:36230))
NODE listening on Some(Tcp(V4(127.0.0.1:49372)))
NOTE: Transport connect [::1]:34818 failure due to Connection refused (os error 111)
Waiting for responses from network
NewConnection on Tcp(V4(127.0.0.1:34228)) while waiting on network.
NewConnection on Tcp(V4(45.55.207.180:34228)) while waiting on network.
NewConnection on Tcp(V4(127.0.0.1:36230)) while waiting on network.
NewConnection on Tcp(V4(45.55.207.180:36230)) while waiting on network.
NewConnection on Tcp(V4(127.0.0.1:34818)) while waiting on network.
NewConnection on Tcp(V4(45.55.207.180:34818)) while waiting on network.
NOTE: Transport connect [::1]:34228 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:36230 failure due to Connection refused (os error 111)
NOTE: Transport connect 45.55.207.180:50704 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:50704 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:50704 failure due to Connection refused (os error 111)
BOOTSTRAP to Tcp(V4(127.0.0.1:34228))
NODE listening on Some(Tcp(V4(127.0.0.1:55538)))
Waiting for responses from network
NewConnection on Tcp(V4(127.0.0.1:34228)) while waiting on network.
NewConnection on Tcp(V4(45.55.207.180:34228)) while waiting on network.
NewConnection on Tcp(V4(127.0.0.1:36230)) while waiting on network.
NewConnection on Tcp(V4(45.55.207.180:36230)) while waiting on network.
NOTE: Transport connect [::1]:34228 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:36230 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:50704 failure due to Connection refused (os error 111)
NOTE: Transport connect 45.55.207.180:50704 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:50704 failure due to Connection refused (os error 111)
BOOTSTRAP to Tcp(V4(127.0.0.1:34228))
NODE listening on Some(Tcp(V4(127.0.0.1:48468)))
thread '<unnamed>' panicked at 'assertion failed: cm.send(bootstrapped_to.clone(), serialised_message).is_ok()', /home/maidsafe/.cargo/registry/src/github.com-0a35038f75765ae4/routing-0.1.64/src/routing_node.rs:159
NOTE: Transport connect [::1]:34228 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:36230 failure due to Connection refused (os error 111)
NOTE: Transport connect 127.0.0.1:50704 failure due to Connection refused (os error 111)
NOTE: Transport connect 45.55.207.180:50704 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:50704 failure due to Connection refused (os error 111)
NOTE: Transport connect [::1]:34818 failure due to Connection refused (os error 111)
BOOTSTRAP to Tcp(V4(45.55.207.180:36230))
NODE listening on Some(Tcp(V4(127.0.0.1:35107)))
Waiting for responses from network
NewConnection on Tcp(V4(127.0.0.1:34228)) while waiting on network.
NewConnection on Tcp(V4(45.55.207.180:34228)) while waiting on network.
NewConnection on Tcp(V4(127.0.0.1:36230)) while waiting on network.
NewConnection on Tcp(V4(45.55.207.180:36230)) while waiting on network.
NewConnection on Tcp(V4(45.55.207.180:34818)) while waiting on network.
NewConnection on Tcp(V4(127.0.0.1:34818)) while waiting on network.
ok
test vault::test::cache_test ... ok
test vault::test::churn_test ... ok
test vault::test::put_get_flow ... ok
test version_handler::test::handle_account_transfer ... ok
test version_handler::test::handle_put_get ... ok
test version_handler::test::version_handler_sendable_serialisation ... ok

failures:

---- test::executable_test stdout ----
        ---------- starting node 0 --------------
---------- starting node 1 --------------
---------- starting node 2 --------------
---------- starting node 3 --------------
---------- starting node 4 --------------
---------- starting node 5 --------------
---------- starting node 6 --------------
---------- starting node 7 --------------
          maidsafe_vault 7 has 0 connected connections in addition to the bootstrapped one.
thread 'test::executable_test' panicked at 'assertion failed: `(left == right)` (left: `8`, right: `2`)', src/main.rs:131



failures:
    test::executable_test

test result: FAILED. 37 passed; 1 failed; 0 ignored; 0 measured

thread '<main>' panicked at 'Some tests failed', /home/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libtest/lib.rs:255
```

**Comment by maqi** [comment]
This is due to the refactor of crust/routing and an improper setup of bootstrap cache file.

**Comment by maqi** [comment]
there is another issue causing this problem : the first node needs to be run in memberane mode.
this needs to be resolved

**Comment by maqi** [comment]
trying to resolve the issue in the PR https://github.com/maidsafe/maidsafe_vault/pull/101

**Comment by maqi** [comment]
resolving in PR https://github.com/maidsafe/maidsafe_vault/pull/104
